File: 백준/Gold/1922. 네트워크 연결/네트워크 연결.py
def find(x):
    if parent[x] == x:
        return x
    parent[x] = find(parent[x])
    return parent[x]

def min_tree():
    result = 0
    for start, end, wei in com:
        # 정점 u와 정점 v에 대한 대표자를 확인
        x = find(start)
        y = find(end)
        if x != y:
            parent[y] = x
            result += wei
    return result

N = int(input())
M = int(input())
com = [list(map(int,input().split())) for _ in range(M)]
com = sorted(com, key= lambda x : x[2])
parent = [i for i in range(N+1)]
print(min_tree())

File: 백준/Gold/1922. 네트워크 연결/네트워크 연결.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    static  int[] parents;
    static int N, M, start, end, wei;
    static List<Node> info = new ArrayList<>();
    static class Node {
        int start;
        int end;
        int wei;
        public Node(int start, int end, int wei) {
            this.start = start;
            this.end = end;
            this.wei = wei;
        }
        public int getWei() {
            return this.wei;
        }
    }
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        N = Integer.parseInt(br.readLine());
        M = Integer.parseInt(br.readLine());
        for (int i = 0; i < M; i++) {
            st = new StringTokenizer(br.readLine());
            start = Integer.parseInt(st.nextToken());
            end = Integer.parseInt(st.nextToken());
            wei = Integer.parseInt(st.nextToken());
            info.add(new Node(start,end,wei));
        }
        Collections.sort(info,Comparator.comparing(Node::getWei));
        parents = new int[N + 1];
        for (int i = 0; i < N + 1; i++) {
            parents[i] = i;
        }
        System.out.println(tree());
    }

    private static int find(int x) {
        if (parents[x] == x) return x;
        parents[x] = find(parents[x]);
        return parents[x];
    }
    private static int tree() {
        int result = 0;
        for (Node node : info) {
            int x = find(node.start);
            int y = find(node.end);
            if (x != y) {
                parents[y] = x;
                result += node.wei;
            }
        }
        return result;
    }
}
